crates/bevy_ecs/src/query/fetch.rs

```diff
@@ -1527,7 +1584,8 @@ macro_rules! impl_anytuple_fetch {
             ) -> Self::Item<'w> {
                 let ($($name,)*) = _fetch;
                 ($(
-                    $name.1.then(|| $name::fetch(&mut $name.0, _entity, _table_row)),
+                    // SAFETY: The invariants are uphold by the caller.
```

Comment:
```suggestion
                    // SAFETY: The invariants are required to be upheld by the caller.
```

This is also sort of incomplete as a safety comment, but given this PR is focused on the lint, I won't block on this.